server/statistics_handler.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pingcap/tidb"
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/model"
+	log "github.com/sirupsen/logrus"
+)
+
+// StatsHandler is the handler for dump statistics.
+type StatsHandler struct {
+	do *domain.Domain
+}
+
+func (s *Server) newStatsHandler() *StatsHandler {
+	store, ok := s.driver.(*TiDBDriver)
+	if !ok {
+		panic("Illegal driver")
+	}
+	do, err := tidb.BootstrapSession(store.store)
+	if err != nil {
+		log.Error("Failed to bootstrap session", err)
+		return nil
+	}
+	return &StatsHandler{do}
+}
+
+func (sh StatsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	params := mux.Vars(req)
+
+	is := sh.do.InfoSchema()
+	h := sh.do.StatsHandle()
+
+	tbl, err := is.TableByName(model.NewCIStr(params[pDBName]), model.NewCIStr(params[pTableName]))
```

Comment:
If TableName is empty, shall we return the info of all tables ?

Reply:
I think `serdes` is more unclear in meaning than `dump`.
Dumping all tables will be added in the future.